FFS-2247: Track user modal interaction events from Pinwheel

tdooner opened the pull request:
<!-- ---------------------------------------------------------------------------
Some examples of good, understandable PR titles:

FFS-1111: Fix missing translation on /entry page
FFS-2222: Implement invitation reminder emails

(The title of the pull request will be used in the eventual deploy log - so it's helpful to format the title to be understandable by other disciplines if possible.)
--------------------------------------------------------------------------- -->
## Ticket

Resolves [FFS-2247](https://jiraent.cms.gov/browse/FFS-2247).


## Changes
<!-- What was added, updated, or removed in this PR. -->
**Refactor "user action" tracking on employer search page**
> * Make the /api/pinwheel/user_action endpoint generic so it can receive
>   an "eventName" and arbitrary "attributes" for an event. This will be
>   used in subsequent commits for Pinwheel modal events.
> * Create new event, `ApplicantSelectedEmployerOrPlatformItem`, to
>   replace events `ApplicantSelectedPopularPayrollPlatform` and
>   `ApplicantSelectedPopularAppEmployer`. A unified event prevents the
>   javascript from having to know which event to send, and will also make
>   funnel analysis and future maintenance easier.
> * Add an attribute `is_default_option` to the new event so we can tell
>   apart clicks on the 12 default items from clicks on search results.
> * Add the `name` attribute to also be present on search results (as it
>   was previously only present on the default options)
> * Add tests for user action API

**Track eight events from Pinwheel modal interactions**
> These events cover most of what we're hoping to instrument for future
> Pinwheel sessions. In the order a user may experience them:

> 1. `PinwheelShowProviderConfirmationPage` - sent when a provider
>    confirmation screen is shown (e.g. "Amazon" showing multiple
>    platforms like A to Z and ADP)
> 2. `PinwheelShowLoginPage` - sent when the first screen of a login is
>    shown (whether it's employer disambiguation, username/password, or
>    other input required by user)
> 3. `PinwheelAttemptLogin` - sent when the user attempts login (sent only
>    once on the first screen even if there are subsequent screens for MFA
>    or other employer-specific questions). Unfortunately, there are no
>    Pinwheel events on the subsequent login pages.
> 4. `PinwheelError` - sent when an error occurs, likely an incorrect
>    login or MFA code, but also a system error.
> 5. `PinwheelSuccess` - sent when the login completes.

> If the user doesn't follow the intended path through the modal, they may
> trigger:

> 6. `PinwheelShowDefaultProviderSearch` - sent when the user deselects
>    the employer or platform and returns to the beginning of the pinwheel
>    modal
> 7. `PinwheelAttemptClose` - shown when the user clicks the "X" in the
>    modal, before confirming the closure
> 8. `PinwheelCloseModal` - shown when the modal actually closes


## Context for reviewers
<!-- Anything you'd like other engineers on the team to know. -->


## Acceptance testing
<!-- Check one: -->

- [ ] No acceptance testing needed
  * This change will not affect the user experience (bugfix, dependency updates, etc.)
- [x] Acceptance testing prior to merge
  * This change can be verified visually via screenshots attached below or by sending a link to a local development environment to the acceptance tester
  * Acceptance testing should be done by **design** for visual changes, **product** for behavior/logic changes, **or both** for changes that impact both.
- [ ] Acceptance testing after merge
  * This change is hard to test locally, so we'll test it in the demo environment (deployed automatically after merge.)
  * Make sure to notify the team once this PR is merged so we don't inadvertently deploy the unaccepted change to production. (e.g. `:alert: Deploy block! @ffs-eng I just merged PR [#123] and will be doing acceptance testing in demo - please don't deploy until I'm finished!`)

GeorgeCodes19 reviewed (approved):
Since we're now consolidating the `ApplicantSelectedPopularPayrollPlatform` and `ApplicantSelectedPopularAppEmployer` I've updated the [Confluence doc with the tracked events](https://confluenceent.cms.gov/display/SFIV/CBV+New+Relic+Events).

tdooner commented:
> I've updated the [Confluence doc with the tracked events](https://confluenceent.cms.gov/display/SFIV/CBV+New+Relic+Events).

Thanks, I forgot about that. I just updated it with the other 8 events as well.

tdooner commented:
Acceptance testing with Eryn:

Renamed event:
- [ ] `ApplicantSelectedEmployerOrPlatformItem`

New events:
- [ ] `PinwheelShowProviderConfirmationPage` - sent when a provider
    confirmation screen is shown (e.g. "Amazon" showing multiple
    platforms like A to Z and ADP)
- [ ] `PinwheelShowLoginPage` - sent when the first screen of a login is
   shown (whether it's employer disambiguation, username/password, or
    other input required by user)
- [ ] `PinwheelAttemptLogin` - sent when the user attempts login (sent only
    once on the first screen even if there are subsequent screens for MFA
    or other employer-specific questions). Unfortunately, there are no
    Pinwheel events on the subsequent login pages.
- [ ] `PinwheelError` - sent when an error occurs, likely an incorrect
   login or MFA code, but also a system error.
- [ ] `PinwheelSuccess` - sent when the login completes.

If the user doesn't follow the intended path through the modal, they may
trigger:

- [ ] `PinwheelShowDefaultProviderSearch` - sent when the user deselects
    the employer or platform and returns to the beginning of the pinwheel
    modal
- [ ] `PinwheelAttemptClose` - shown when the user clicks the "X" in the
    modal, before confirming the closure
- [ ] `PinwheelCloseModal` - shown when the modal actually closes
